### ISSUE TITLE
Crypt::PK::ECC - Apple's APNS pkcs8 auth key import fails

### ISSUE THREAD

**Issue by karel-m** [DCIT/perl-CryptX#28]
Original report by @rockbone

I'm  developping a library for sending notification via APNS using Apple's new API.
I have to make a request using a private key obtained from Apple, but I can not import that key with Crypt :: PK :: ECC.

In openssl command(OpenSSL 1.0.2j  26 Sep 2016), I could import by doing as follows.
```
    my $secret = `openssl pkcs8 -nocrypt -in APNs.p8`;
```
In Crypt :: PK :: ECC, I did the following.
```
    my $c = new Crypt::PK::ECC;
    $c->import_key("APNs.p8");
```

And error messages are following.
```
    # FATAL: ecc_import_pkcs8 failed: Invalid input packet.
```

**Comment by karel-m** [comment]
I confirm that it can be reproduced by:
```
perl -MCrypt::PK::ECC -MData::Dumper -E 'print Dumper(Crypt::PK::ECC->new("APNs.p8")->key2hash)'
```

The file `APNs.p8` is in PEM format (probably generated by Apple).

The workaround is to convert `APNs.p8` by openssl from PEM to PEM (not a typo) - which basically means loading the PEM key by openssl and re-saving it as another PEM (which was in my case a bit smaller than the original .p8 file):
```
openssl pkcs8 -in APNs.p8 -inform PEM -out APNs.pem -outform PEM -nocrypt 
```

After that you can load the new `APNs.pem` like this:
```
perl -MCrypt::PK::ECC -MData::Dumper -E 'print Dumper(Crypt::PK::ECC->new("APNs.pem")->key2hash)'
```

So obviously in Apple's PEM there is "something" which Crypt::PK::ECC module does not expect and throws an exception (`Invalid input packet` means `I do not understand the structure of input data`).

I'll try to investigate more.

**Comment by karel-m** [comment]
This is ASN.1 structure of the original key from Apple (Crypt::PK::ECC **is not** capable to load it):
```
Offset|Length|LenByte|
======+======+=======+==============================
     0|   147|      2|   SEQUENCE : 
     3|     1|      1|      INTEGER : '0'
     6|    19|      1|      SEQUENCE : 
     8|     7|      1|         OBJECT_IDENTIFIER : 'ECC (1.2.840.10045.2.1)'
    17|     8|      1|         OBJECT_IDENTIFIER : 'ECDSA_P256 (1.2.840.10045.3.1.7)'
    27|   121|      1|      OCTET_STRING : 
    29|   119|      1|         SEQUENCE : 
    31|     1|      1|            INTEGER : '1'
    34|    32|      1|            OCTET_STRING : 
      |      |       |               052...private.key...
    68|    10|      1|            CONTEXT SPECIFIC (0) : 
    70|     8|      1|               OBJECT_IDENTIFIER : 'ECDSA_P256 (1.2.840.10045.3.1.7)'
    80|    68|      1|            CONTEXT SPECIFIC (1) : 
    82|    66|      1|               BIT_STRING : 
      |      |       |                  Unused bits: 0 : 043...public.key...
```

And this is the one re-saved by openssl (Crypt::PK::ECC **is** capable to load it)
```
Offset|Length|LenByte|
======+======+=======+==============================
     0|   119|      1|   SEQUENCE : 
     2|     1|      1|      INTEGER : '1'
     5|    32|      1|      OCTET_STRING : 
      |      |       |         052...private.key...
    39|    10|      1|      CONTEXT SPECIFIC (0) : 
    41|     8|      1|         OBJECT_IDENTIFIER : 'ECDSA_P256 (1.2.840.10045.3.1.7)'
    51|    68|      1|      CONTEXT SPECIFIC (1) : 
    53|    66|      1|         BIT_STRING : 
      |      |       |            Unused bits: 0 : 043...public.key...
```

It means that `src/ltc/pk/ecc/ecc_import_pkcs8.c` needs to be extended.

**Comment by karel-m** [comment]
The conclusion of my analysis:
* `Crypt::PK::ECC` supports pkcs8 keys according https://tools.ietf.org/html/rfc5208
* however there exists newer pkcs8 standard https://tools.ietf.org/html/rfc5958 which `Crypt::PK::ECC` currently does not support

**Comment by karel-m** [comment]
Should be fixed in CryptX 0.058_001 

I'll close this issue after 0.059 release

**Comment by karel-m** [comment]
fixed in 0.059
